pkg/localkube/apiserver.go

```diff
@@ -99,7 +99,10 @@ func StartAPIServer(lk LocalkubeServer) func() error {
 }
 
 func readyFunc(lk LocalkubeServer) HealthCheck {
-	hostport := net.JoinHostPort("localhost", strconv.Itoa(lk.APIServerPort))
-	addr := "https://" + path.Join(hostport, "healthz")
-	return healthCheck(addr, lk)
+	protocol := "https://"
+	if lk.APIServerInsecurePort != 0 {
+		protocol = "http://"
```

Comment:
I think you generally want to stay away from constructions like
```
default value
if condition {
 overwrite default value
}
```

Here is another place where you're implicitly hiding the behavior of 
**"if both secure and insecure serving are enabled, use the insecure serving port"**
which is a bit arbitrary and I'd like to minimize the amount of places that logic has to live, so its easy to change/maintain/notice

Reply:
This could live in the GetTransport function - since if you're adding a TLSConfig, you know that you'll be using `https`

Reply:
done